src/gui/servernotificationhandler.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) by Klaas Freitag <freitag@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "servernotificationhandler.h"
+#include "accountstate.h"
+#include "capabilities.h"
+#include "json.h"
+#include "networkjobs.h"
+
+namespace OCC
+{
+
+ServerNotificationHandler::ServerNotificationHandler(QObject *parent)
+    : QObject(parent)
+{
+
+}
+
+void ServerNotificationHandler::slotFetchNotifications(AccountState *ptr)
+{
+    /* start the notification fetch job as well */
+    if( !ptr) {
+        return;
+    }
+
+    // check if the account has notifications enabled. If the capabilities are
+    // not yet valid, its assumed that notifications are available.
+    if( ptr->account() && ptr->account()->capabilities().isValid() ) {
+        if( ! ptr->account()->capabilities().notificationsAvailable() ) {
+            qDebug() << "Account" << ptr->account()->displayName() << "does not have notifications enabled.";
```

Comment:
Please remove or use qDebug() << Q_FUNC_INFO consistent with the rest of the general code base. We should probably clean this up.